File: Backend/AstroGame.Shared/Models/Resources/ResourceSnapshot.cs
using AstroGame.Shared.Models.Stellar.BaseTypes;
using System;
using System.Collections.Generic;

namespace AstroGame.Shared.Models.Resources
{
    public class ResourceSnapshot
    {
        public Guid Id { get; set; }
        public Guid StellarObjectId { get; set; }
        public DateTime SnapshotTime { get; set; }
        public List<StoredResource> StoredResources { get; set; } = new();

        public virtual StellarObject StellarObject { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/resources/resource_snapshot.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/resources/stored_resource.dart';

import 'package:json_annotation/json_annotation.dart';

part 'resource_snapshot.g.dart';

@GuidConverter()
@JsonSerializable()
class ResourceSnapshot {
  late Guid id;
  late Guid stellarObjectId;
  late DateTime snapshotTime;
  List<StoredResource> storedResources = [];

  ResourceSnapshot();

  factory ResourceSnapshot.fromJson(Map<String, dynamic> json) =>
      _$ResourceSnapshotFromJson(json);
  Map<String, dynamic> toJson() => _$ResourceSnapshotToJson(this);
}
